A --no-html project tries to "npm install" but can't find the "phoenix_html" dependency.

When I start a new project with Phoenix 1.1.2 with the `--no-html` option, I get the following:
```
...
Fetch and install dependencies? [Yn]
* running mix deps.get
* running npm install && node node_modules/brunch/bin/brunch build

We are all set! Run your Phoenix application:
...
```
Starting the server `mix phoenix.server` gives the following error:
```
[error] Could not start node watcher because script "node_modules/brunch/bin/brunch" does not exist. Your Phoenix application is still running, however assets won't be compiled. You may fix this by running "npm install".
```

Doing `npm install` gives the following error:
```
npm ERR! addLocal Could not install /Users/abdullah/Desktop/elixir/project/deps/phoenix_html
npm ERR! Darwin 15.2.0
npm ERR! argv "/usr/local/Cellar/node/5.3.0/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v5.3.0
npm ERR! npm  v3.3.12
npm ERR! path /Users/abdullah/Desktop/elixir/project/deps/phoenix_html
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall open

npm ERR! enoent ENOENT: no such file or directory, open '/Users/abdullah/Desktop/elixir/project/deps/phoenix_html'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/abdullah/Desktop/elixir/preject/npm-debug.log
```

Good catch. It should simply be a matter of skipping it in brunch-config
and package.json.

On Sunday, January 10, 2016, Abdullah Esmail <notifications@github.com>
wrote:

> When I start a new project with Phoenix 1.1.2 with the --no-html option,
> I get the following:
>
> ...
> Fetch and install dependencies? [Yn]
> * running mix deps.get
> * running npm install && node node_modules/brunch/bin/brunch build
>
> We are all set! Run your Phoenix application:
> ...
>
> Starting the server mix phoenix.server gives the following error:
>
> [error] Could not start node watcher because script "node_modules/brunch/bin/brunch" does not exist. Your Phoenix application is still running, however assets won't be compiled. You may fix this by running "npm install".
>
> Doing npm install gives the following error:
>
> npm ERR! addLocal Could not install /Users/abdullah/Desktop/elixir/project/deps/phoenix_html
> npm ERR! Darwin 15.2.0
> npm ERR! argv "/usr/local/Cellar/node/5.3.0/bin/node" "/usr/local/bin/npm" "install"
> npm ERR! node v5.3.0
> npm ERR! npm  v3.3.12
> npm ERR! path /Users/abdullah/Desktop/elixir/project/deps/phoenix_html
> npm ERR! code ENOENT
> npm ERR! errno -2
> npm ERR! syscall open
>
> npm ERR! enoent ENOENT: no such file or directory, open '/Users/abdullah/Desktop/elixir/project/deps/phoenix_html'
> npm ERR! enoent This is most likely not a problem with npm itself
> npm ERR! enoent and is related to npm not being able to find a file.
> npm ERR! enoent
>
> npm ERR! Please include the following file with any support request:
> npm ERR!     /Users/abdullah/Desktop/elixir/preject/npm-debug.log
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1465>.
>


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

--- a/installer/templates/static/brunch/brunch-config.js
+++ b/installer/templates/static/brunch/brunch-config.js
@@ -65,6 +65,6 @@ exports.config = {
     enabled: true,
     // Whitelist the npm deps to be pulled in as front-end assets.
     // All other deps in package.json will be excluded from the bundle.
-    whitelist: ["phoenix", "phoenix_html"]
+    whitelist: ["phoenix"<%= if html do %>, "phoenix_html"<% end %>]
   }
 };

--- a/installer/templates/static/brunch/package.json
+++ b/installer/templates/static/brunch/package.json
@@ -8,7 +8,7 @@
     "css-brunch": ">= 1.0 < 1.8",
     "javascript-brunch": ">= 1.0 < 1.8",
     "uglify-js-brunch": ">= 1.0 < 1.8",
-    "phoenix": "file:<%= brunch_deps_prefix %><%= phoenix_path %>",
-    "phoenix_html": "file:<%= brunch_deps_prefix %>deps/phoenix_html"
+    "phoenix": "file:<%= brunch_deps_prefix %><%= phoenix_path %>"<%= if html do %>,
+    "phoenix_html": "file:<%= brunch_deps_prefix %>deps/phoenix_html"<% end %>
   }
 }

